src/dune/process.ml

```diff
@@ -2,23 +2,13 @@ open! Stdune
 open Import
 open Fiber.O
 
-type accepted_codes =
-  | These of int list
-  | All
-
-let code_is_ok accepted_codes n =
-  match accepted_codes with
-  | These set -> List.mem n ~set
-  | All -> true
-
 type ('a, 'b) failure_mode =
   | Strict : ('a, 'a) failure_mode
-  | Accept : accepted_codes -> ('a, ('a, int) result) failure_mode
+  | Accept : (int -> bool) -> ('a, ('a, int) result) failure_mode
```

Comment:
Is there an issue with circular dependencies that doesn't let us supply the predicate lang here directly?

Reply:
No, no issue, I amended as suggested. Thanks!